client/src/components/view-prompts.tsx

```diff
@@ -0,0 +1,162 @@
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+
+import React, { useContext } from "react";
+
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogTitle,
+  IconButton,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemText,
+  TextField,
+} from "@mui/material";
+import { Close as CloseIcon } from "@mui/icons-material";
+import CogenerationContext from "context/cogeneration";
+export interface SimpleDialogProps {
+  open: boolean;
+  selectedPrompt: string;
+  onClose: (value: string) => void;
+  setSelectedPrompt: React.Dispatch<React.SetStateAction<string>>;
+}
+
+function ViewPrompts(props: SimpleDialogProps): JSX.Element {
+  const context = useContext(CogenerationContext);
+  if (!context) {
+    throw new Error("SomeComponent must be used within a CogenerationProvider");
+  }
+  const { onClose, selectedPrompt, open, setSelectedPrompt } = props;
+
+  const handleClose = () => {
+    onClose(selectedPrompt);
+  };
+
+  const getDefaultValues = (type: string) => {
+    if (type === "multipleChoice") {
+      return { prompt: "", systemPrompt: "" };
+    } else if (type === "lesson") {
+      return {
+        systemPrompt: "",
+        humanPrompt: "",
+        parseSystemPrompt: "",
+        parseHumanPrompt: "",
+        parseSystem2: "",
+        parseHuman2: "",
+        tabooSystemPrompt: "",
+        tabooHumanPrompt: "",
+        simulationSystemPrompt: "",
+        simulationHumanPrompt: "",
+        slotsSystemPrompt: "",
+        slotsHumanPrompt: "",
+        genericSystemPrompt: "",
+        genericHumanPrompt: "",
+        hintFilterSystemPrompt: "",
+        hintFilterHumanPrompt: "",
+        hintMockPrompt: "",
+        patchSystemPrompt: "",
+        patchHumanPrompt: "",
+      };
+    }
+    return {};
+  };
+
+  let selectedSet = { title: "error", type: "error404" };
+
+  if (context.generationData.genRecipe === "multipleChoice") {
+    selectedSet = context.generationData.MCQPrompts.find(
+      (pair) => pair.title === selectedPrompt
+    ) || {
+      title: "error",
+      type: "error404",
+      ...getDefaultValues("multipleChoice"),
+    };
+  } else {
+    selectedSet = context.generationData.lessonPrompts.find(
+      (pair) => pair.title === selectedPrompt
+    ) || { title: "error", type: "error404", ...getDefaultValues("lesson") };
+  }
+
+  const renderTextFieldsFromObject = (fieldsObject: Record<string, string>) => {
+    return Object.keys(fieldsObject).map((fieldKey) => (
+      <TextField
+        key={fieldKey}
+        data-cy={`${fieldKey}-output`}
+        label={fieldKey
```

Comment:
whatever these replace values are doing might be useful to have as a separate helper functions (name of the function should also help let other developers know what it's doing)